### PR TITLE
Handle data manager last sync error in the syncComponent method

### PR DESCRIPTION
**Pull request by vbihun**
# Description

Please include a description of changes introduced in this PR.

# Checklist

Please ensure that each of these items has been addressed:

- [x] I have tested these changes in my local environment
- [x] I have added/modified tests as applicable to cover these changes
- [x] (Atlassian contributors only) I have removed any Atlassian-internal changes including internal modules, references to internal tickets, and internal wiki links

### CODE REVIEW

File: src/utils/push-event-utils.ts [hunk 1]
```diff
@@ -43,3 +44,8 @@ export const isEventForTrackingBranch = (event: PushEvent, trackingBranch: strin
 
   return event.ref === trackingBranchRef;
 };
+
+export const isUpdateCompassComponentDataManager = (
```
Comment (eserranovalle) [inline src/utils/push-event-utils.ts:48]: If this utility only needs to check whether this contains a `lastSyncEvent`, a `has` prefix in the method name could clarify.

`hasUpdateCompassComponentDataManagerLastSyncEvent` could be a mouthful. Maybe Maybe `hasLastSyncEvent`? The typed parameter should clarify what this applies to.

File: src/utils/push-event-utils.test.ts [hunk 2]
```diff
@@ -57,3 +58,24 @@ describe('groupDiffsByChangeType', () => {
     expect(result).toEqual(expectedResult);
   });
 });
+
+describe('isUpdateCompassComponentDataManager', () => {
+  it('should returns true in case if dataManager has lastSyncEvent', () => {
```
Comment (eserranovalle) [inline src/utils/push-event-utils.test.ts:63]: returns -> return

File: src/utils/push-event-utils.test.ts [hunk 3]
```diff
@@ -57,3 +58,24 @@ describe('groupDiffsByChangeType', () => {
...
+      componentId: 'id',
+      lastSyncEvent: { lastSyncErrors: ['error'], status: ComponentSyncEventStatus.UserError },
+    };
+
+    const result = isUpdateCompassComponentDataManager(dataManager);
+
+    expect(result).toBe(true);
+  });
+
+  it('should returns false in case if dataManager has no lastSyncEvent', () => {
```
Comment (eserranovalle) [inline src/utils/push-event-utils.test.ts:74]: returns -> return

File: src/utils/push-event-utils.ts (hunk 1 above)
Comment (vbihun) [reply src/utils/push-event-utils.ts:48]: Good catch! Thank you @eserranovalle

**Review by eserranovalle — APPROVED** [review]
Looks good!

**Pull request merged**